testsuite/classlibrary/TestPprotect.sc

```diff
@@ -19,4 +20,75 @@ TestPprotect : UnitTest {
 		cond.hang;
 		this.assert(success, "Pprotect should clear the stream's exceptionHandler");
 	}
+
+	test_stream_can_be_restarted_after_error {
+		var pat, stream, hasRun = false;
+		pat = Pprotect(
+			Prout {
+				0.01.yield;
+				hasRun = true;
+				Error("dummy error").throw
+			},
+			{ stream.streamError }
+		);
+		stream = pat.play;
+		0.02.wait;
+		hasRun = false;
+		stream.reset;
+		stream.play;
+		0.02.wait;
+		this.assert(hasRun, "stream should be resettable after an error");
+	}
+
+	test_task_proxy_play_after_error {
+		var proxy, redefine, hasRun;
+		proxy = TaskProxy.new;
+		proxy.quant = 0;
+		proxy.play;
+
+		redefine = {
+			proxy.source = {
+				0.01.wait;
+				hasRun = true;
+				Error("dummy error").throw
+			}
+		};
+		hasRun = false;
+		redefine.value;
+		0.02.wait;
+		hasRun = false;
+		redefine.value;
+		0.02.wait;
+		this.assert(hasRun, "task proxy should play again after an error");
+	}
+
+
+
+	// this stops the test run, calling test_nested_instances twice, works.
+	// keeping ths commented out for now
+
+	test_nested_instances {
+
+		fork {
```

Comment:
same as with the TestErrorHandling test

Reply:
I want this newline here, it looks better.